sample-apps/payment-customization-sample-app/extensions/hide-payment-by-name-and-cart-subtotal/src/main.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+
+mod api;
+use api::*;
+
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Configuration {
+    pub payment_method: String,
+    #[serde_as(as = "DisplayFromStr")]
+    pub cart_subtotal: Decimal,
+}
+
+impl Configuration {
+    fn from_str(value: &str) -> Self {
+        serde_json::from_str(value).expect("Unable to parse configuration value from metafield")
+    }
+}
+
+impl Input {
+    pub fn configuration(&self) -> Option<Configuration> {
+        self.payment_customization.metafield.as_ref()
+            .map(|metafield| Configuration::from_str(&metafield.value))
+    }
+}
+
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /* TODO: after the patterns team adds the `presentment_currency_rate` field to the Input we need to change the following:
+        1. remove `mut` from the line bellow
+        2. remove the assignment to presentment_currency_rate = 1.0 on line 40
+        3. add presentmentCurrencyRate to input.graphql
+    */
```

Comment:
We should remove this TODO before we ship this. If this can't be fixed before shipping, then please split this into a separate task to tackle in the future.